File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    for i in goal:
        if len(cards1) > 0 and cards1[0] == i:
            del cards1[0]
            continue
        if len(cards2) > 0 and cards2[0] == i:
            del cards2[0]
            continue
        return "No"
    return "Yes"

File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    while(goal.length !== 0){
        const word = goal.shift();
        if(cards1[0] === word){
            cards1.shift();
            continue;
        } else if(cards2[0] === word){
            cards2.shift();
            continue;
        } else{
            return "No";
        }
    };
    return "Yes";
}